lib/sudo-pam/src/rpassword.rs

```diff
@@ -7,35 +7,36 @@
 ///
 /// Most code was replaced and so is no longer a derived work; work that we kept:
 ///
-/// - the "HiddenInput" struct and implementation
+/// - the "HiddenInput" struct and implementation, with changes:
 ///   * replaced occurences of explicit 'i32' and 'c_int' with RawFd
-///   * make it return an Option ("None" if the given fd is not a terminal)
+///   * open the TTY ourselves to mitigate Linux CVE-2023-2002
 /// - the general idea of a "SafeString" type that clears its memory
 ///   (although much more robust than in the original code)
 ///
 use std::io::{self, Read};
-use std::os::fd::{AsFd, AsRawFd, RawFd};
+use std::os::fd::{AsRawFd, RawFd};
 use std::{fs, iter, mem};
 
-use libc::{isatty, tcsetattr, termios, ECHO, ECHONL, TCSANOW};
+use libc::{tcsetattr, termios, ECHO, ECHONL, TCSANOW};
```

Comment:
sort of unrelated but why `sudo-pam` doesn't use `sudo-system`?

Reply:
That's just due to evolution. `rpassword.rs` wasn't the result of some grand design; `sudo-pam` used to depend on the rpassword crate and the `HiddenInput` is the vestigial remnant of that. I can't think of an obstacle to move that particular type definition over to `sudo-system` (at the same time, I don't think currently there is a *need* for it elsewhere)

Reply:
Meaning, right now `HiddenInput` is not `pub`, moving it to sudo-system would mean it is `pub` and it doesn't need to be that.

Reply:
oh what I mean is that I thought that we canonically added all the `libc` wrappers we use to `sudo-system`.

Reply:
`sudo-cutils` had that intent originally. Thinking along that line, we could write a more general safe wrapper around the "termios" structure and the `tcsetattr` function. That would be a good thing to do if we discover we need to scratch that itch in two unrelated places I think.